src/regtech_data_validator/check_functions.py

```diff
@@ -318,7 +318,10 @@ def is_number(ct_value: str, accept_blank: bool = False) -> bool:
     Returns:
         bool: True if value is number , False if value is not number
     """
-    value_check = ct_value.isdigit() or bool(re.match(r"^[-+]?[0-9]*\.?[0-9]+$", ct_value))
+    if is_whole:
+        value_check = ct_value.isdigit()
+    else:
+        value_check = ct_value.isdigit() or bool(re.match(r"^[-+]?[0-9]*\.?[0-9]+$", ct_value))
```

Comment:
I know the regex bits were already there, but something like this seems like simpler logic.

```suggestion
    value_check = True
    num_parser = int if is_whole else float
    try:
        num_parser(ct_value)
    except ValueError:
        value_check = False
```

Also, `str.isdigit` probably isn't the best check. You can put funky stuff in there like:

```python
>>> '2²'.isdigit()
True
```

I think `str.isdecimal` is the more appropriate function...if we want to stick with that pattern:

```python
>>> '2²'.isdecimal()
False
>>> '2.0'.isdecimal()
False
>>> '2'.isdecimal()
True
```